modules/common/annotations/network.go

```diff
@@ -54,3 +55,44 @@ func GetNADAnnotation(namespace string, nads []string) (map[string]string, error
 
 	return map[string]string{NetworkAttachmentAnnot: string(networks)}, nil
 }
+
+// GetAnnotation - returns an annotation value based on the key passed as input
+// This function also returns true of false based on the fact that a value associated
+// to the key passed as input exists. Returning a boolean is useful to allow wrapping
+// this function in calls like `value, ok := GetAnnotation(ann, key); ok {` and
+// take actions based on the existance of the entry in the annotation map[string]string
+func GetAnnotation(
+	ann map[string]string,
+	key string,
+) (string, bool) {
+	if value, ok := ann[key]; ok {
+		return value, ok
+	}
+	return "", false
+}
+
+// GetBoolFromAnnotation - it returns a boolean from a string annotation
+// e.g. glance.openstack.org/wsgi: "true" returns true as a boolean type
+//
+// Cases covered by this function:
+// 1. the annotation does not exist -> false, false, nil
+// 2. the annotation exist and is not a valid boolean -> false, true, error
+// 3. the annotation exists and is a valid False bool -> false, true, nil
+// 4. the annotation exists and is a valid True bool -> true, true, nil
+func GetBoolFromAnnotation(
+	ann map[string]string,
+	key string,
+) (bool, bool, error) {
+	result, exists := false, false
+	var value string
+	var err error
+	// Try to get the value associated to the annotation key
+	if value, exists = GetAnnotation(ann, key); exists {
+		result, err = strconv.ParseBool(value)
+		if err != nil {
+			// the annotation is not a valid boolean, return an error
+			return false, exists, err
+		}
+	}
+	return result, exists, nil
```

Comment:
I guess we could simplify to this, or am I missing a case?
```
value, exists := ann[key] 
if !exists {
    return false, false, nil
}

result, err := strconv.ParseBool(value)
if err != nil {
	return false, true, err
}

return result, true, nil
```

Reply:
I assume the current version is equivalent to what you have there. I simply don't do an explicit `!exists` check because the variable is already set by `GetAnnotation` that returns true or false (the last return covers the !exists part)

Reply:
Done